util/src/util/Database.ts

```diff
@@ -1,72 +1,103 @@
 import path from "path";
 import "reflect-metadata";
-import { Connection, createConnection } from "typeorm";
+import { DataSource, createConnection, DataSourceOptions, PrimaryColumn, PrimaryGeneratedColumn } from "typeorm";
 import * as Models from "../entities";
 import { Migration } from "../entities/Migration";
 import { yellow, green, red } from "picocolors";
+import fs from "fs";
+import { exit } from "process";
+import { BaseClass, BaseClassWithoutId } from "../entities";
+import { config } from "dotenv";
 
 // UUID extension option is only supported with postgres
 // We want to generate all id's with Snowflakes that's why we have our own BaseEntity class
 
-var promise: Promise<any>;
-var dbConnection: Connection | undefined;
-let dbConnectionString = process.env.DATABASE || path.join(process.cwd(), "database.db");
+let promise: Promise<any>;
+let dataSource: DataSource;
 
-export function initDatabase(): Promise<Connection> {
-	if (promise) return promise; // prevent initalizing multiple times
+export async function initDatabase(): Promise<DataSource> {
+	//if (dataSource) return dataSource; // prevent initalizing multiple times
 
-	const type = dbConnectionString.includes("://") ? dbConnectionString.split(":")[0]?.replace("+srv", "") : "sqlite";
-	const isSqlite = type.includes("sqlite");
+	if(dataSource.isInitialized) return dataSource;
```

Comment:
Ahhhh that initDb call in lazyrequest makes sense now I guess. Although it would've made more sense if this function ( or a new function ) had a better name now

Reply:
Resolved.